src/Spice86.Core/Emulator/VM/Breakpoint/BreakpointConditionCompiler.cs

```diff
@@ -0,0 +1,52 @@
+namespace Spice86.Core.Emulator.VM.Breakpoint;
+
+using Spice86.Core.Emulator.CPU;
+using Spice86.Core.Emulator.CPU.CfgCpu.Ast.Parser;
+using Spice86.Core.Emulator.CPU.CfgCpu.Ast.Value;
+using Spice86.Core.Emulator.CPU.CfgCpu.InstructionExecutor.Expressions;
+using Spice86.Core.Emulator.Memory;
+using System.Linq.Expressions;
+
+/// <summary>
+/// Compiles breakpoint condition expressions into executable functions using Expression trees.
+/// This class is shared across different parts of the codebase (GDB, UI, serialization)
+/// to provide consistent condition evaluation with high performance via compiled native code.
+/// </summary>
+public class BreakpointConditionCompiler {
+    private readonly State _state;
+    private readonly Memory _memory;
```

Comment:
Accept an instance of IIndexable instead. Propagate this to AstExpressionBuilder and remove the check in the constructor when done.